crates/polars-plan/src/logical_plan/optimizer/projection_pushdown/projection.rs

```diff
@@ -8,6 +8,24 @@ fn is_count(node: Node, expr_arena: &Arena<AExpr>) -> bool {
     }
 }
 
+/// Make sure that the rolling key is projected
+#[cfg(feature = "dynamic_group_by")]
+fn add_rolling_key(
+    node: Node,
+    expr_arena: &mut Arena<AExpr>,
+    acc_projections: &mut Vec<Node>,
+    projected_names: &mut PlHashSet<Arc<str>>,
+) {
+    if let AExpr::Window {
```

Comment:
I think we should use `has_aexpr` here. That way we traverse the whole expression tree and ensure we always add it if needed.